### PR TITLE
Re-add deferred tests

### PR DESCRIPTION
**Pull request by wolfs**
And fix tests.

### CODE REVIEW

File: .teamcity/Gradle_Check/projects/StageProject.kt [hunk 1]
```diff
@@ -62,14 +62,16 @@ class StageProject(model: CIBuildModel, stage: Stage, containsDeferredTests: Boo
             id = AbsoluteId(uuid)
             name = "Test coverage deferred from Quick Feedback and Ready for Merge"
             model.buildTypeBuckets
-                .filter { it.shouldBeSkippedInStage(stage) }
+                .filter { !it.shouldBeSkippedInStage(stage) && it.containsSlowTests() }
```
Comment (blindpirate) [inline .teamcity/Gradle_Check/projects/StageProject.kt]: According to https://github.com/gradle/gradle/blob/ddb80cb79cc882e42be704deb9b3d280db6e8aa5/.teamcity/Gradle_Check/projects/StageProject.kt#L66

Only `it.containsSlowTests()` is necessary here - we're trying to add slow test subproject into later stage.

File: .teamcity/Gradle_Check/projects/FunctionalTestProject.kt [hunk 2]
```diff
@@ -22,15 +21,8 @@ class FunctionalTestProject(model: CIBuildModel, testConfig: TestCoverage, stage
...
-            // https://github.com/gradle/gradle-private/issues/2693
-            if (testConfig.testType.supportTestSplit) {
-                buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name, bucket.extraParameters()))
-            } else if (bucket.number == 1) {
-                buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name))
-            }
-        } else {
-            buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name, bucket.extraParameters()))
+        for (buildTypeBucket in bucket.forTestType(testConfig.testType)) {
+            buildType(FunctionalTest(model, testConfig, buildTypeBucket.getSubprojectNames(), stage, buildTypeBucket.name, buildTypeBucket.extraParameters()))
```
Comment (blindpirate) [inline .teamcity/Gradle_Check/projects/FunctionalTestProject.kt]: Note that when we're not splitting the project, we should not add `-testSplit=1/3` parameter.

Comment (wolfs) [reply .teamcity/Gradle_Check/projects/FunctionalTestProject.kt]: And we don't: if we are not splitting the project, then `bucket.forTestType` will return a buildTypeBucket with `extraParameters == ''`.

Comment (blindpirate) [reply .teamcity/Gradle_Check/projects/FunctionalTestProject.kt]: OK, yes you're right, it returns `listOf(subproject)`, not `listOf(this)`.

**Pull request merged**